metagraph/src/annotation/binary_matrix/column_sparse/column_major.cpp

```diff
@@ -126,6 +132,12 @@ uint64_t ColumnMajor::num_relations() const {
     return num_set_bits;
 }
 
+void ColumnMajor::call_columns(const ColumnCallback &callback) {
+    for(uint64_t i = 0; i < data_.size(); ++i) {
+        callback(i, std::move(data_[i]));
+    }
+}
```

Comment:
Why not just return a const reference to `data_`?

Reply:
ah, I missed that it moves them. In this case, return the object itself `std::vector<...>`.

The implementation is simpler. Also, maybe in some cases, the target thing you need is this vector. So if we implement it with a callback, you'd have to put all these bit vectors back to an std::vector in the callback, which is a bit stupid because they were initially in an std::vector.

So the idea here is to reduce the total number of conversions which may happen.
If we return the entire `data_`, we will never do extra overhead for sure.
Essentially, this loop `for ...` will be moved to `annotation_converters.cpp`, where you need to construct a callback.